qiskit/primitives/containers/observables_array.py

```diff
@@ -116,12 +132,18 @@ def tolist(self) -> list | ObservableLike:
                 >>> print(type(oa.tolist()))
                 <class 'dict'>
         """
-        return self._array.tolist()
+        return self.__array__().tolist()
 
-    def __array__(self, dtype=None, copy=None):
-        """Convert to an Numpy.ndarray"""
+    def __array__(self, dtype=None, copy=None) -> np.ndarray:  # pylint: disable=unused-argument
```

Comment:
You might want to raise an error if `copy=False` like https://numpy.org/doc/stable/user/basics.dispatch.html#writing-custom-array-containers because this method seems to make a copy always.

Reply:
Is it possible to avoid a copy with `copy=False`? Then, it's fine as is.

> copy: bool, optional
> For False it raises a ValueError if a copy cannot be avoided. Default: None.
https://numpy.org/doc/stable/reference/generated/numpy.asarray.html#numpy.asarray

Reply:
Yes, it creates new objects, which are retuned.